### ISSUE TITLE
Constant SslCloseCompletionEvent logging

### ISSUE THREAD

**Issue by skypartan** [aws/aws-sdk-java-v2#2867]
### Describe the issue

I'm part of the development team of a moderately utilized web application. We are currently using S3 to store all users files, however we've found that there is a lot of spam on the log files about exceptions in channels that are not beign utilized. I've already checked all the places in the code where the api is utilized, making sure that all open streams were properly closed, so I discarded that option. I tried changing the HTTP backend to Aws Crt Http however there was a severe ram usage increase, so we changed back. The netty backend presented some more issues. We have little experience with AWS Apis, and have not been able to find helpful information regarding this issue.

### Steps to Reproduce

We have not been able to find exactly what is causing the problem, either in the source code or in the execution environment. We belive it to be internal operations of the S3 client library as all stacktraces show internal packages.

### Current behavior

The following message is spammed to the log files constantly, only in production where the are a lot more people using the system:

`[2021-11-23T11:22:34.296-0300] [Payara 5.2021.8] [WARNING] [] [software.amazon.awssdk.http.nio.netty.internal.UnusedChannelExceptionHandler] [tid: _ThreadID=365 _ThreadName=aws-java-sdk-NettyEventLoop-1-2] [timeMillis: 1637677354296] [levelValue: 900] [[
  A non-I/O exception occurred on a channel (ab25f9b9) that was not in use. The channel has been closed to prevent any ongoing issues.
java.lang.NoClassDefFoundError: io/netty/handler/ssl/SslCloseCompletionEvent
        at software.amazon.awssdk.http.nio.netty.internal.SslCloseCompletionEventHandler.userEventTriggered(SslCloseCompletionEventHandler.java:51)
        at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:339)
        at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:325)
        at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:317)
        at io.netty.handler.ssl.SslHandler.setHandshakeSuccess(SslHandler.java:1193)
        at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1008)
        at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:900)
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:411)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:248)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:129)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:610)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:551)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:465)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:437)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
        at java.lang.Thread.run(Thread.java:748)
]]`

### AWS Java SDK version used

2.17.23

### JDK version used

Temeriun 1.8.0_312-b07

### Operating System and version

Red Hat Enterprise Linux release 8.4 (Ootpa)

**Comment by debora-ito** [comment]
Hi @skypartan thank you for the detailed report.

```
java.lang.NoClassDefFoundError: io/netty/handler/ssl/SslCloseCompletionEvent at 
software.amazon.awssdk.http.nio.netty.internal.SslCloseCompletionEventHandler.userEventTriggered(SslCloseCompletionEventHandler.java:51)
```

`NoClassDefFoundError` usually indicates that a class was present during compile time and the application was compiled and linked successfully, but the class was not present during runtime.

The `SslCloseCompletionEvent` class is from the `io.netty:netty-handler` dependency, check if the class is present in your classpath. If it is, take a look at the answers in this StackOverflow post, they show other possible root causes - 
https://stackoverflow.com/questions/17973970/how-can-i-solve-java-lang-noclassdeffounderror

**Comment by skypartan** [comment]
After some searching throught the classpath I found out that the netty-handler version that was being included is a previous version to the one awssdk utilizes, probably included by another project depency. I checked the classes in the library and found out that the included version did not have the SslCloseCompletionEvent class, causing the error. I've forced maven to utilize the new version of the library and looks like everything is working. Thanks for the help!
